pkg/client/kubectl_client.go

```diff
@@ -448,3 +466,31 @@ func (k *KubectlResourceOperations) authReconcile(obj *unstructured.Unstructured
 	}
 	return strings.Join(out, ". "), nil
 }
+
+type replacedStdin struct {
+	orig *os.File
+}
+
+func pumpToStdin(data io.Reader) (*replacedStdin, error) {
+	stdinRead, stdinWrite, err := os.Pipe()
+	if err != nil {
+		return nil, err
+	}
+
+	f := &replacedStdin{orig: os.Stdin}
+
+	os.Stdin = stdinRead
```

Comment:
again , I'm not sure I follow why we need this . Could we point `stdinRead` to `*os.File` directly ? 

Also does this mean it's using a file for storing the data or it's using the stdin stream?